### PR TITLE
Add CallTreeViewItemModelTest

### PR DESCRIPTION
**Pull request by antonrohr**

### CODE REVIEW

File: src/OrbitQt/CallTreeViewItemModelTest.cpp [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+#include "ClientData/PostProcessedSamplingData.h"
+#include "ClientModel/CaptureData.h"
+#include "ClientModel/SamplingDataPostProcessor.h"
+#include "QtUtils/AssertNoQtLogWarnings.h"
+#include "capture.pb.h"
+#include "capture_data.pb.h"
+
+constexpr uint64_t kCallstackId = 1;
+constexpr uint64_t kAbsoluteFunctionAddress = 0x30;
+constexpr uint64_t kAbsoluteInstructionAddress = 0x31;
```
Comment (dpallotti) [inline src/OrbitQt/CallTreeViewItemModelTest.cpp:22]: I would suggest `kFunctionAbsoluteAddress` and `kInstructionAbsoluteAddress`.

File: src/OrbitQt/CallTreeViewItemModelTest.cpp [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  // AddressInfo
+  orbit_client_protos::LinuxAddressInfo address_info;
+  address_info.set_absolute_address(kAbsoluteInstructionAddress);
+  address_info.set_offset_in_function(kAbsoluteInstructionAddress - kAbsoluteFunctionAddress);
+  address_info.set_function_name(kFunctionName);
+  address_info.set_module_path(kModuleName);
+  capture_data->InsertAddressInfo(address_info);
+
+  // CallstackInfo
+  const std::vector<uint64_t> callstack_frames{kAbsoluteInstructionAddress};  // , 0x21, 0x11};
```
Comment (dpallotti) [inline src/OrbitQt/CallTreeViewItemModelTest.cpp:43]: The trailing comment is a leftover?

File: src/OrbitQt/CallTreeViewItemModelTest.cpp [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+  orbit_client_data::PostProcessedSamplingData sampling_data =
+      orbit_client_model::CreatePostProcessedSamplingData(*capture_data->GetCallstackData(),
+                                                          *capture_data, false);
+
+  auto call_tree_view =
+      CallTreeView::CreateTopDownViewFromPostProcessedSamplingData(sampling_data, *capture_data);
+
+  CallTreeViewItemModel model{std::move(call_tree_view)};
+
+  // 1 top level entry for thread with id: kThreadId
```
Comment (dpallotti) [inline src/OrbitQt/CallTreeViewItemModelTest.cpp:130]:
```suggestion
  // One top level entry for thread with id: kThreadId
```
Just my OCD.

File: src/OrbitQt/CallTreeViewItemModelTest.cpp [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
```
Comment (dpallotti) [inline src/OrbitQt/CallTreeViewItemModelTest.cpp:215]:
```suggestion
  // One top level entry for thread with id: kThreadId
```

**Review by dpallotti — APPROVED** [review]
Nice!

**Review by beckerhe — APPROVED** [review]

File: src/OrbitQt/CallTreeViewItemModelTest.cpp (hunk 2 above)
Comment (antonrohr) [reply src/OrbitQt/CallTreeViewItemModelTest.cpp]: Yes it is. Thank you!

**Pull request merged**